### ISSUE TITLE
Regression: God of War 3 massive speed loss on Intel Xeon CPU's

### ISSUE THREAD

**Issue by Linear524** [RPCS3/rpcs3#11918]
Strange speed issues occur for God of War 3 after commit 95ac772 was applied, or maybe after it was merged to master branch (I mean maybe some other thing was changed in master branch during that day...)
Overall speed was decreased by half after first intro video was loaded.

![BCES00510_screenshot_2022_05_02_21_51_06](https://user-images.githubusercontent.com/84367301/166412506-61420e5b-ed35-4361-94ec-4797d488ef41.jpg)
![BCES00510_screenshot_2022_05_02_21_51_11](https://user-images.githubusercontent.com/84367301/166412510-f4c43682-795e-4cad-b0ad-f4343f33fd8d.jpg)
![BCES00510_screenshot_2022_05_02_21_51_29](https://user-images.githubusercontent.com/84367301/166412513-6546ab9a-5f5d-48bb-8d50-ecfc1dff1382.jpg)

I've tried to run this on my second PC and same issue happens there too. I'm using Intel Xeon E5 2690 V4 and Xeon E5 2680 V3 CPU's. Before that commit my speed was 100% in menus (60fps) and 45~59 fps ingame, now it is always just a half.

My main rig:
SYS: Intel(R) Xeon(R) CPU E5-2690 v4 @ 3.20GHz | 14 Cores | 31.83 GiB RAM |
SYS: Operating system: Windows, Major: 10, Minor: 0, Build: 18362
GPU: 'NVIDIA GeForce GTX 1080' running on driver 471.41.0.0
SYS: Firmware version: 4.87

My settings:
WCB+RCB = ON
LLVM Recompilers.
SPU block size = Mega
Zcull Accuracy = Fast.
FSR = 50%
PPU Non-Java mode fixup = ON
TSX = OFF or Forced (no difference)
No external patches applied.
Everything else is set to Default.

**Comment by Linear524** [comment]
Here is my Capture data:
[BCES00510_20220502215141_capture.zip](https://github.com/RPCS3/rpcs3/files/8608678/BCES00510_20220502215141_capture.zip)

**Comment by kd-11** [comment]
Why such an old gpu driver?

**Comment by Linear524** [comment]
I'm using this particular GPU driver because it gives best image quality in my homebrew VR application for Oculus Rift S...
But is it necessary to update for latest in case of RPCS3 ? I'm using studio version drivers, not DCH versions.
(I never had any issues with other titles on 471.41)

**Comment by kd-11** [comment]
Well, rpcs3 does not like driver versions 470.0 -> 511.x and there are still bugs nvidia is fixing with the new driver branch, so I would suggest upgrading. However, the report itself is incomplete and non-conclusive.
Get screenshots and logs comparing the before and after as they were added to the emulator.
Build 1 (before): https://github.com/RPCS3/rpcs3-binaries-win/releases/download/build-5746df4046915e959928945d372b07e1501af06f/rpcs3-v0.0.21-13528-5746df40_win64.7z
BUild 2 (after): https://github.com/RPCS3/rpcs3-binaries-win/releases/download/build-8316469cfcfb076d5a9772557b0de522f39616d2/rpcs3-v0.0.21-13532-8316469c_win64.7z

To summarize:
1. Install latest driver (you can revert after the test data is captured)
2. Get the new screenshots and logs and attach to the ticket. That way other users can try to reproduce.

**Comment by Linear524** [comment]
Ok ! Will test tonight :>

**Comment by Linear524** [comment]
Holy shit...
I've checked your builds and discovered that problem sits deeper, actually - it was PR11849 (CPU optimizations) which broke down my speed on all my Xeon V3 and Xeon V4 CPUs... All that CPU's are TSX-FA family and that PR was changed some wait engine behavior... 
But I've tried disabling TSX in my every test, and the biggest discovery for me right now was actual require of enabling Accurate RSX reservation access after after PR11849 for God of War 3... Noone talk about this little thing after that PR was merged to master.
So right now if I enable Accurate RSX reservation access - my speed is back to 100% and gameplay runs normal again (*_*)
I will make some further testing right now with high-load cases and close this isssue :)

P.S. - maybe some other people should know about requirement of Accurate RSX reservation access for God of War 3 after PR11849 for TSX-FA CPUs...

**Comment by Linear524** [comment]
Megamouse, thanks ! I didn't know that. But I never used Accurate RSX until now. GoW3 was always at fullspeed on my rig before...
I guess this title have some specific tight timings
